scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +171,369 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
+    >>> dist = genexpon(9, 16, 3)  # freeze the distribution
+    >>> p = np.linspace(0.01, 0.99, 99)  # percentiles from 1% to 99%
+    >>> time_once(lambda: dist.ppf(p))
+    '154.565 ms'  # may vary
+
+    >>> time_once(lambda: dist.rvs(size=100))
+    '148.979 ms'  # may vary
+
+    The `FastNumericalInverse` has a method that approximates ``dist.ppf``.
+
+    >>> from scipy.stats import FastNumericalInverse
+    >>> fni = FastNumericalInverse(dist)
+    >>> np.allclose(fni.ppf(p), dist.ppf(p))
+    True
+
+    In some cases, it is faster to both generate the fast numerical inverse
+    and use it than to call ``dist.ppf``.
+
+    >>> def time_me():
+    ...     fni = FastNumericalInverse(dist)
+    ...     fni.ppf(p)
+    >>> time_once(time_me)
+    '11.9222 ms'  # may vary
+
+    After generating the fast numerical inverse, subsequent calls to its
+    methods are much faster.
+    >>> time_once(lambda: fni.ppf(p))
+    '0.0819 ms'  # may vary
+
+    The fast numerical inverse can also be used to generate random variates
+    using inverse transform sampling.
+
+    >>> time_once(lambda: fni.rvs(size=100))
+    '0.0911 ms'  # may vary
+
+    Depending on the implementation of the distribution's random sampling
+    method, the random variates generated may be nearly identical, given
+    the same random state.
+
+    >>> rvs1 = dist.rvs(size=100, random_state=0)
+    >>> rvs2 = fni.rvs(size=100, random_state=0)
+    >>> np.allclose(rvs1, rvs2)
+    True
+
+    """
+
+    def __init__(self, dist, tol=1e-12, max_intervals=100000):
+        H, eu, intervals = _fast_numerical_inverse(dist, tol, max_intervals)
+        self.H = H
+        self.midpoint_error = eu
+        self.intervals = intervals
+
+    def ppf(self, q):
+        r"""
+        Approximate percent point function (inverse `cdf`) of the given RV.
+
+        Parameters
+        ----------
+        q : array_like
+            lower tail probability.
+
+        Returns
+        -------
+        x : array_like
+            quantile corresponding to the lower tail probability `q`.
+
+        """
+        q = np.asarray(q)  # no harm; self.H always returns an array
+        result = np.zeros_like(q, dtype=np.float64)
+        i = (q >= 0) & (q <= 1)
+        result[i] = self.H(q[i])
+        result[~i] = np.nan
+        return result
+
+    def rvs(self, size=None, random_state=None):
+        """
+        Random variates of the given RV.
+
+        The `random_state` is used to draw uniform pseudo-random variates, and
+        these are converted to pseudo-random variates of the given RV using
+        inverse transform sampling.
+
+        Parameters
+        ----------
+        size : int, tuple of ints, or None; optional
+            Defines shape of array of random variates. Default is ``None``.
+        random_state : {None, int, `numpy.random.Generator`,
+                        `numpy.random.RandomState`}, optional
```

Comment:
I think this is what's causing the Sphinx complaint.

Reply:
It's because in this case it needs a blank space due to the 2 lines description. I got bitten by that as well 😉

Reply:
Thanks! I always forget.